pkg/workceptor/kubernetes.go

```diff
@@ -226,23 +226,42 @@ func podRunningAndReady(kw KubeUnit) func(event watch.Event) (bool, error) {
 	return inner
 }
 
-func GetTimeoutOpenLogstream(kw *KubeUnit) int {
-	// RECEPTOR_OPEN_LOGSTREAM_TIMEOUT
-	// default: 1
-	openLogStreamTimeout := 1
-	envTimeout := os.Getenv("RECEPTOR_OPEN_LOGSTREAM_TIMEOUT")
+func (kw *KubeUnit) GetKubeTimeoutStart() time.Duration {
+	// RECEPTOR_KUBE_TIMEOUT_START
+	// default: 1 second
+	kubeTimeoutStart := 1 * time.Second
+	envTimeout := os.Getenv("RECEPTOR_KUBE_TIMEOUT_START")
 	if envTimeout != "" {
 		var err error
-		openLogStreamTimeout, err = strconv.Atoi(envTimeout)
-		if err != nil || openLogStreamTimeout < 1 {
+		kubeTimeoutStart, err = time.ParseDuration(envTimeout)
+		if err != nil || kubeTimeoutStart <= 0 {
 			// ignore error, use default
-			kw.GetWorkceptor().nc.GetLogger().Warning("Invalid value for RECEPTOR_OPEN_LOGSTREAM_TIMEOUT: %s. Ignoring", envTimeout)
-			openLogStreamTimeout = 1
+			kw.GetWorkceptor().nc.GetLogger().Warning("Invalid value for RECEPTOR_KUBE_TIMEOUT_START: %s. Ignoring", envTimeout)
+			kubeTimeoutStart = 1 * time.Second
 		}
 	}
-	kw.GetWorkceptor().nc.GetLogger().Debug("RECEPTOR_OPEN_LOGSTREAM_TIMEOUT: %d", openLogStreamTimeout)
+	kw.GetWorkceptor().nc.GetLogger().Debug("RECEPTOR_KUBE_TIMEOUT_START: %s", kubeTimeoutStart)
 
-	return openLogStreamTimeout
+	return kubeTimeoutStart
+}
+
+func (kw *KubeUnit) GetKubeRetryCount() int {
+	// RECEPTOR_KUBE_RETRY_COUNT
+	// default: 5
+	kubeRetryCount := 5
+	envRetryCount := os.Getenv("RECEPTOR_KUBE_RETRY_COUNT")
+	if envRetryCount != "" {
+		var err error
+		kubeRetryCount, err = strconv.Atoi(envRetryCount)
+		if err != nil || kubeRetryCount < 1 {
```

Comment:
Should there be an upper bound on the retry count? A high number plus a fib backoff could mean very long wait times if the retry count is set to something like 100.